pkg/util/winutil/pdhutil/pdhcounter.go

```diff
@@ -108,20 +110,38 @@ func (counter *pdhCounter) ShouldInit() bool {
 		// already initialized
 		return false
 	}
+	var initFailLimit = config.Datadog.GetInt("windows_counter_init_failure_limit")
+	if initFailLimit > 0 && counter.initFailCount >= initFailLimit {
+		counter.initError = fmt.Errorf("Counter exceeded the maximum number of failed initialization attempts. This error indicates that the Windows performance counter database may need to be rebuilt.")
+		// attempts exceeded
+		return false
+	}
 	return true
 }
 
+func (counter *pdhCounter) countInitError(err error) error {
+	counter.initFailCount += 1
+	var initFailLimit = config.Datadog.GetInt("windows_counter_init_failure_limit")
```

Comment:
Ah, yeah, I have no idea myself. Check in the source if the conversion happens. There is probably explicit config call. On other other hand do not twist in pretzel because it is good enough.